lineapy/api/api.py

```diff
@@ -140,43 +141,71 @@ def save(reference: object, name: str) -> LineaArtifact:
     return linea_artifact
 
 
-def delete(
-    artifact_name: str, version: Optional[Union[int, str]] = None
-) -> None:
+def delete(artifact_name: str, version: Union[int, str]) -> None:
     """
     Deletes an artifact from artifact store. If no other artifacts
     refer to the value, the value is also deleted from both the
     value node store and the pickle store.
 
-    If version is not provided, latest version is used.
+    :param artifact_name: Key used to while saving the artifact
+    :param version: version number or 'latest' or 'all'
     """
+    if not is_artifact_version_valid(version):
+        raise ValueError(
+            f"Version {version} is not a valid version.\n"
+            + "Version must be a positive whole number or 'all' or 'latest'."
+        )
+
+    # if version is an integer string, cast to int
+    if version not in ["all", "latest"]:
+        version = int(version)
```

Comment:
More specifically, we could change the function to `try_parse_version` and return `Optional[int]`, where if the value is `None`, we have a parsing error, otherwise it's the int for the version.